proposals/1859_github-auth/README.md

```diff
@@ -0,0 +1,500 @@
+---
+title: "Authenticating Users via GitHub"
+authors: [ "@cfryanr" ]
+status: "draft"
+sponsor: [ ]
+approval_date: ""
+---
+
+*Disclaimer*: Proposals are point-in-time designs and decisions.
+Once approved and implemented, they become historical documents.
+If you are reading an old proposal, please be aware that the
+features described herein might have continued to evolve since.
+
+# Authenticating Users via GitHub
+
+## Problem Statement
+
+Many developers in the world have a GitHub account, and many enterprises use GitHub.
+This makes GitHub a convenient identity provider for an enterprise to use to control
+access to Kubernetes clusters for its developers.
+
+This document proposes adding a GitHubIdentityProvider resource to the Pinniped Supervisor,
+which would allow GitHub to be used as an identity provider for authentication to Kubernetes
+clusters.
+
+### How Pinniped Works Today (as of version v0.28.0)
+
+The Pinniped Supervisor currently supports OIDC-compliant identity providers and
+LDAP identity providers (including Active Directory). Unfortunately, GitHub does not offer
+either of these protocols for authenticating users. GitHub uses a slightly customized version
+of OAuth 2.0 for customizing users, so Pinniped will need to add new features to support GitHub.
```

Comment:
It may be helpful here to expand on what Github's customized OAuth 2.0 app is.  I'm thinking about the distinction, OAuth is about authorization, not authentication.  We are really using it for authentication to a kubernetes cluster, and still assuming users will use Kube RBAC for authorization of abilities within the cluster.

Reply:
Added a paragraph to clarify this.